caput/tests/test_mpiarray.py

```diff
@@ -583,14 +581,11 @@ def test_reduce():
 
         assert (df_total == df_sum.allreduce()).all()
 
-        df_total = np.zeros_like(df_sum)
-
-        df_sum.Allreduce(df_total)
-
-        assert (df_total == 4 * size).all()
-
-        assert df_total.axis == 0
+        # Test MPIArray.allreduce(); ensure it returns
+        # an ndarray vector with appropriate values + shape
+        assert (dist_array.allreduce() == 4).all()
+        assert (dist_array.allreduce()).shape == (4, 1)
 
-        with pytest.raises(ValueError):
-            # ValueError should be raised, since df_total and dist_array are not the correct shape
-            dist_array.Allreduce(df_total)
+        # MPIArray.sum().allreduce() should give the scalar sum of
+        # all entries
+        assert df_sum.allreduce() == 4 * size
```

Comment:
I don't think this will work if the number of elements on each rank is different, i.e.
```
assert comm.size == 4
dist_arr = MPIArray((5,), comm=comm)

dist_arr.allreduce()
```
So I would suggest adding a test for it and seeing if it will break.

Reply:
That case does not work with the regular `Allreduce` either!

You get: `mpi4py.MPI.Exception: MPI_ERR_TRUNCATE: message truncated` with both situations.

How would you like the code to handle this? Perform a check, and print a more readable error message?

Reply:
From a zoom call:

the case we are most interested in is `array.sum().allreduce()`, so we are deciding to restrict usage of `allreduce()` to arrays with a single element.